test_discovery.py

```diff
@@ -160,39 +160,57 @@ def test_single_entity(entities, test_name, test_value):
     return (0, 0)
 
 
-def test_single_case(test):
+def test_single_case(test_dict, response_intent_dict):      #most_likely_intent
     """
     Run a single test case
     Return the number of errors
-    """
-    print("======\nTesting: {}".format(test['test']))
-    resp = submit_transcript(test['transcript'])
 
-    # Check if a valid response was received
-    if not is_valid_response(resp):
-        fail_test(resp)
+    Compares expected entities (from test) with discovery returned entities
+
+    :param test_dict: dict;
+        key: str; name of entity
+        value: str; first occurrence of entity in transcript
+
+    :param response_intent_dict: dict, single intent from Discovery response
+
+    :return: 4-Tuple
+       first: 0 if tests pass, 1 if tests fails
 
-    # For now, only keep the first intent:
-    intent = resp["intents"][0]
+       total_errors: number of entities in test whose observed value (str) differs from expected
+
+       total_char_errors: sum of number of char that differ between observed and expected values for each entity
+
+       characters:
+    """
+    # print("======\nTesting: {}".format(test['test']))
+    # resp = submit_transcript(test['transcript'])
+    #
+    # # Check if a valid response was received
```

Comment:
How come we're removing this check?

Reply:
this was moved to `test_all` so the intent can be checked first and, if the label returned differs from the expected intent, then entity testing for that intent is skipped